src/TextFieldRich.jsx

```diff
@@ -3,234 +3,439 @@ var UAParser = require('ua-parser-js');
+			
+			// retrieve selected range
+			var sel = this._getIframeDoc().selection; 
+			if (sel != null) {
+				var newRng = sel.createRange();
+				newRng = this._rangeSelection;
+				newRng.select();
+			}
+		}
+	},
+
+	/**
+	 * Handles the input blur event of the text area
+	 *
+	 * @param {DOMEvent} e		Reference to the DOM event being sent 
+	 * @private
+	 */
+	_handleInputBlur: function(e) {
+
+		if (typeof e.target === "undefied") {
+			e.target = {};
+		}
+			
+		e.target.value = this.getValue();
+
+		if (this.props.onBlur) this.props.onBlur(e);
+	},
+
+	/**
+	 * Handles the input change event of the text area
+	 *
+	 * @param {DOMEvent} e		Reference to the DOM event being sent 
+	 * @private
+	 */
+	_handleInputChange: function(e) {
+		if (this.props.onChange) this.props.onChange(e);
+	},
+
+	/**
+	 * Handles the input focus of the text area
+	 *
+	 * @param {DOMEvent} e		Reference to the DOM event being sent 
+	 * @private
+	 */
+	_handleInputFocus: function(e) {
+		if (this.props.onFocus) {
+			this.props.onFocus(e);
+		}
+	},
+
+	/**
+	 * Handles the input key down of the text area
+	 *
+	 * @param {DOMEvent} e		Reference to the DOM event being sent 
+	 * @private
+	 */
+	_handleInputKeyDown: function(e) {
+		if (this.props.autoGrow) {
+			this._autoGrow();
+		}
+
+		if (e.keyCode === 13 && this.props.onEnterKeyDown) {
+			this.props.onEnterKeyDown(e);
+		}
     
-    if (this.isMounted()) {
-      this._autoGrow();
-    }
-  },
-
-  _handleInputBlur: function(e) {
-
-    if (typeof e.target === "undefied")
-      e.target = {};
-    e.target.value = this.getValue();
-
-    this._handleInputChange(e);
-
-    if (this.props.onBlur) this.props.onBlur(e);
-
-  },
-
-  _handleInputChange: function(e) {
-    if (this.props.onChange) this.props.onChange(e);
-  },
-
-  _handleInputFocus: function(e) {
-    this.setState({isFocused: true});
-    if (this.props.onFocus) this.props.onFocus(e);
-  },
-
-  _handleInputKeyDown: function(e) {
-    if (this.props.autoGrow) {
-      this._autoGrow();
-    }
-
-    if (e.keyCode === 13 && this.props.onEnterKeyDown) {
-      this.props.onEnterKeyDown(e);
-    }
+		// Handle onkeydown event
+		if (this.props.onKeyDown) {
+			this.props.onKeyDown(e);
+		}
+		
+		// Handle onChange event
+		if (this.props.onChange) {
+			this.props.onChange(e);
+		}
+	},
+
+	/**
+	 * Handles the change of height event of the text area
+	 *
+	 * @param {DOMEvent} e		Reference to the DOM event being sent 
+	 * @param {int} height	T	The height to be set
+	 * @private
+	 */
+	_handleTextAreaHeightChange: function(e, height) {
+		var newHeight = height + 24;
+		if (this.props.floatingLabelText) {
+			newHeight += 24;
+		}
+		this.getDOMNode().style.height = newHeight + 'px';
+	},
+
+	/**
+	 * Check if the component is controlled
+	 * 
+	 * @private
+	 */
+	_isControlled: function() {
+		return this.props.hasOwnProperty('value') || this.props.hasOwnProperty('valueLink');
+	},
+
+	/**
+	 * Get the iframe window/document
```

Comment:
Please change the comments to this for better clarity: 

* Get the iframe content window/document
* 
* IE < version 8 does not supply contentWindow so we also have to check for contentDocument.